test/sql/test_temporal.py

```diff
@@ -0,0 +1,221 @@
+            period,
+        )
+        eq_(t.application_versioning, False)
+        eq_(t.system_versioning, False)
+        is_(period, t.c.test_period)
+        eq_(period.name, "test_period")
+        eq_(period.start, "start")
+        eq_(period.end, "end")
+
+
+class ApplicationVersioningTest(fixtures.TestBase, AssertsCompiledSQL):
+    """Application versioning does not currently have anything separate
+    from the Period construct outside of the table's metadata."""
+
+    __dialect__ = "default"
+
+    def test_to_metadata(self):
+        period = Period("test_period", "start", "end")
+        m = MetaData()
+        t = Table(
+            "t",
+            m,
+            Column("start", TIMESTAMP),
+            Column("end", TIMESTAMP),
+            period,
+        )
+        eq_(t.application_versioning, True)
+        eq_(t.system_versioning, False)
+        is_(period, t.c.test_period)
+        eq_(period.name, "test_period")
+        eq_(period.start, "start")
+        eq_(period.end, "end")
+
+
+class SystemVersioningTest(fixtures.TestBase, AssertsCompiledSQL):
+    """Test possible constructs related to system versioning.
+
+    Tests come from MariaDB's implementation examples
+    https://mariadb.com/kb/en/system-versioned-tables/
+    And from "Temporal features in SQL:2011"
+    https://cs.ulb.ac.be/public/_media/teaching/infoh415/
+    tempfeaturessql2011.pdf"""
+
+    __dialect__ = "default"
+
+    # @testing.requires.system_versioned_tables_support
+    def test_create_table_versioning_no_columns(self):
+        m = MetaData()
+        t = Table("t", m, Column("x", Integer), SystemTimePeriod())
+        self.assert_compile(
+            schema.CreateTable(t),
+            "CREATE TABLE t (x INTEGER) WITH SYSTEM VERSIONING",
+        )
+
+    # @testing.requires.system_versioned_tables_support
+    def test_create_table_versioning_columns_specified(self):
+        m = MetaData()
+        t = Table(
+            "t",
+            m,
+            Column("x", Integer),
+            Column("start_timestamp", TIMESTAMP),
+            Column("end_timestamp", TIMESTAMP),
+            SystemTimePeriod("start_timestamp", "end_timestamp"),
+        )
+        self.assert_compile(
+            schema.CreateTable(t),
+            "CREATE TABLE t ("
+            "x INTEGER, "
+            "start_timestamp TIMESTAMP GENERATED ALWAYS AS ROW START, "
+            "end_timestamp TIMESTAMP GENERATED ALWAYS AS ROW END, "
+            "PERIOD FOR SYSTEM_TIME (start_timestamp, end_timestamp)"
+            ") WITH SYSTEM VERSIONING",
+        )
+
+    def test_column_with_system_versioning(self):
+        m = MetaData()
+        t = Table(
+            "t",
+            m,
+            Column("x", Integer, system_versioning=True),
+            Column("y", Integer),
+        )
+        self.assert_compile(
+            schema.CreateTable(t),
+            "CREATE TABLE t ("
+            "x INTEGER WITH SYSTEM VERSIONING,"
```

Comment:
which syntax is this?

Reply:
That's what I referenced here https://github.com/sqlalchemy/sqlalchemy/pull/7933#discussion_r852326473 from the page https://mariadb.com/kb/en/system-versioned-tables/#excluding-columns-from-versioning